src/quick.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding utf-8 -*-
+"""Implementation of a quick sort."""
+import random
+import timeit
+
+def quick(input_list):
+    """Quick sort implementation."""
+    if not isinstance(input_list, list):
+        raise TypeError
+
+    if len(input_list) <= 1:
+        return input_list
+
+    pivot = input_list[-1]
+
+    if len(input_list) == 2:
+        if input_list[0] > pivot:
+            input_list[0], input_list[-1] = input_list[-1], input_list[0]
+        return input_list
+
+    left = 0
+    right = len(input_list) - 2
+
+    while left <= right:
+        if input_list[left] < pivot and left <= len(input_list) - 2:
+            left += 1
+            continue
+        if input_list[right] >= pivot and right >= 0:
+            right -= 1
+            continue
+        if input_list[left] > input_list[right]:
+            input_list[right], input_list[left] = input_list[left], input_list[right]
+
+    input_list[left], input_list[-1] = input_list[-1], input_list[left]
+
+    input_list[:left] = quick(input_list[:left])
+    input_list[left + 1:] = quick(input_list[left + 1:])
+
+    return input_list
```

Comment:
Unless I'm mistaken, it looks like your Quicksort implementation is an in-place implementation, so you don't need to return the same list.

Generally you want either an in-place operation which returns None, or an operation which returns a new data structure. You as the developer want to be clear which option you are using, and more importantly you want your user to be clear.